libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3553,37 +3553,44 @@ string YulUtilFunctions::readFromMemoryOrCalldata(Type const& _type, bool _fromC
 		}
 
 		solAssert(_type.isValueType(), "");
-		if (auto const* funType = dynamic_cast<FunctionType const*>(&_type))
-			if (funType->kind() == FunctionType::Kind::External)
-				return Whiskers(R"(
-					function <functionName>(memPtr) -> addr, selector {
-						let combined := <load>(memPtr)
-						addr, selector := <splitFunction>(combined)
```

Comment:
The change here is a bit hard to see because I combined the two cases again. What was missing was a call to `<validate>(combined)` before this line, but only when reading from calldata.